scripts/oneoff/generate-g-cloud-services.py

```diff
@@ -0,0 +1,163 @@
+6) Runs the `index-services.py` script to index the new services
+"""
+
+import argparse
+import getpass
+import random
+import subprocess
+import sys
+sys.path.insert(0, '.')
+
+from dmapiclient import DataAPIClient, SearchAPIClient  # noqa
+from dmscripts.helpers.env_helpers import get_api_endpoint_from_stage  # noqa
+from dmscripts.json_schema_service_faker import JsonSchemaServiceFaker, gen_gcloud_name  # noqa
+from dmscripts.helpers.service_faker_helpers import eligible_g9_declaration_base
+
+LOTS_WHITELIST = ['cloud-hosting', 'cloud-software', 'cloud-support']
+G9_LOTS = LOTS_WHITELIST
+
+
+def get_args():
+    a = argparse.ArgumentParser()
+
+    a.add_argument('--new-slug', '--new', '-n', type=str, default='g-cloud-9',
+                   help="The (new) g-cloud framework slug to generate services against (default=g-cloud-9)")
+
+    a.add_argument('--old-slug', '--old', '-o', type=str, default='g-cloud-8',
+                   help="The (old) g-cloud framework to use to find suppliers for new services (default=g-cloud-8)")
+
+    a.add_argument('--env', '--stage', '-e', type=str, default='local',
+                   help="Which environment to target for service ingestion (default=local)")
+
+    a.add_argument('--data-api-token', '-d', type=str, default='myToken',
+                   help="The data API token for the selected stage (default=myToken)")
+
+    a.add_argument('--search-api-token', '-s', type=str, default='myToken',
+                   help="The search API token for the selected stage (default=myToken)")
+
+    a.add_argument('--lots', type=str, default=','.join(G9_LOTS),
+                   help="Comma-separated list of lots to generate services for (default={})".format(','.join(G9_LOTS)))
+
+    a.add_argument('quantity', type=int,
+                   help="How many services to generate")
+
+    return a.parse_args()
+
+
+def filepath_service_lot_schema(new_slug, chosen_lot):
+    return 'schemas/services-{}-{}.json'.format(new_slug, chosen_lot)
+
+if __name__ == '__main__':
+    args = get_args()
+
+    args.env = args.env.lower()
+    if args.env not in ['dev', 'development', 'local', 'preview', 'staging']:
+        print("This script can only be run against dev/preview/staging environments.")
+        sys.exit(1)
+
+    args.lots = args.lots.lower().split(',')
+    if set(args.lots) - set(LOTS_WHITELIST):
+        print("This script only allows the following lots: {}. If you need other lots, please add them to the "
+              "whitelist (this is just a sanity-check against typos).".format(LOTS_WHITELIST))
+        sys.exit(1)
+
+    data_api_url = get_api_endpoint_from_stage(args.env, 'api')
+    data_api_client = DataAPIClient(data_api_url, args.data_api_token)
+
+    search_api_url = get_api_endpoint_from_stage(args.env, 'search-api')
+    search_api_client = SearchAPIClient(search_api_url, args.search_api_token)
+
+    services_generated = 0
+    jsfd = JsonSchemaServiceFaker()
```

Comment:
Oh. Yes. That probably is a bit confusing. Think I called it Json Schema Fake Data for a while... Will fix it.